File: emb/core_task/src/types/measurement.cpp
#include "measurement.h"

#include <tuple>

namespace types {
bool operator==(const Measurement& lhs, const Measurement& rhs) {
  return std::make_tuple(lhs.header, lhs.encoder_1_pos, lhs.encoder_2_pos) ==
         std::make_tuple(rhs.header, rhs.encoder_1_pos, rhs.encoder_2_pos);
}
}  // namespace types

File: src/motor_platform/motor_platform/types/measurement.py
from typing import NamedTuple

from motor_platform.types.header import Header


class Measurement(NamedTuple):
    header: Header
    encoder_1_pos: int
    encoder_2_pos: int

    @staticmethod
    def decode(json_value):
        return Measurement(
            header=Header.decode(json_value["header"]),
            encoder_1_pos=json_value["encoder_1_pos"],
            encoder_2_pos=json_value["encoder_2_pos"],
        )
